crates/bevy_ecs/src/reflect.rs

```diff
@@ -27,6 +27,61 @@ pub struct ReflectComponent {
     copy: fn(&World, &mut World, Entity, Entity),
 }
 
+/// The raw function pointers needed to make up a [`ReflectComponent`].
+///
+/// This is used when creating custom implementations of [`ReflectComponent`] with
+/// [`ReflectComponent::new()`].
+///
+/// > **Note:**
+/// > Creating custom implementations of [`ReflectComponent`] is an advanced feature that most users
+/// > will not need.
+/// > Usually you will obtain a [`ReflectComponent`] for a type by retrieving it from the
+/// > [`TypeRegistry`][bevy_reflect::TypeRegistry].
```

Comment:
I would maybe also note how these are generated (i.e. typically using `#[reflect(Component)]`). Same for `ReflectResourceFns`, as well as their respective constructors.